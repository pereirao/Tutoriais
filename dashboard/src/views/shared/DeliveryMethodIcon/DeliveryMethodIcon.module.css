.deliveryMethodIcon {
  vertical-align: middle;
  margin-right: 8px;
}