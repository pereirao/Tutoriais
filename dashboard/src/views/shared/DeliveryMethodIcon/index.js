//@flow

import React from 'react';
import classNames from 'classnames/bind';
import I18n from '../../../localization';

import style_sheet from './DeliveryMethodIcon.module.css';

import pickup from './assets/bag.png';
import shipped from './assets/box.png';
import on_demand from './assets/truck.png';

const cx = classNames.bind(style_sheet);
const icon_map = {pickup, shipped, on_demand};

type DeliveryMethodIconProps = { type: string, inline: boolean };
const DeliveryMethodIcon = ({type, inline = false}: DeliveryMethodIconProps) => (
  <img
    src={icon_map[type]}
    className={cx('deliveryMethodIcon', {inline})}
    alt={I18n.t(`accessibility.icon.${type}`)}/>
);

export default DeliveryMethodIcon;
