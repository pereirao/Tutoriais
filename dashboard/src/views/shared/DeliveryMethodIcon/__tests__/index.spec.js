import React from 'react';

import DeliveryMethodIcon from '../index';

it('renders shipped correctly', () => {
  expect(
    shallow(<DeliveryMethodIcon type="shipped" />)
  ).toMatchSnapshot();
});

it('renders pickup correctly', () => {
  expect(
    shallow(<DeliveryMethodIcon type="pickup" />)
  ).toMatchSnapshot();
});

it('renders on_demand correctly', () => {
  expect(
    shallow(<DeliveryMethodIcon type="on_demand" />)
  ).toMatchSnapshot();
});
