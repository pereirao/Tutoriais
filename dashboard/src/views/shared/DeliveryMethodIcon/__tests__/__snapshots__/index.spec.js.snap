// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders on_demand correctly 1`] = `
<img
  alt="Delivery Order"
  className="deliveryMethodIcon"
  src="truck.png"
/>
`;

exports[`renders pickup correctly 1`] = `
<img
  alt="Pickup Order"
  className="deliveryMethodIcon"
  src="bag.png"
/>
`;

exports[`renders shipped correctly 1`] = `
<img
  alt="Shipping Order"
  className="deliveryMethodIcon"
  src="box.png"
/>
`;
